src/MultiPartStreamFactory.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace Http\Message;
+
+use Psr\Http\Message\StreamInterface;
+
+/**
+ * Factory for creating Multipart Streams.
+ *
+ * @author Robert Schönthal <robert.schoenthal@gmail.com>
+ */
+interface MultiPartStreamFactory
+{
+    /**
+     * Creates a Multipart Stream from parameters and/or files.
+     *
+     * @param array $parameters
+     * @param array $files
+     *
+     * @return StreamInterface
+     *
+     * @throws \InvalidArgumentException If the $parameters or $files are invalid.
```

Comment:
> it would be nice to be a bit more precise. maybe we should also use an exception from the php-http namespace?

Not sure, it is not closely part of the HTTP workflow.